pkg/registry/core/pod/strategy.go

```diff
@@ -153,6 +153,10 @@ func (podStatusStrategy) PrepareForUpdate(ctx genericapirequest.Context, obj, ol
 	oldPod := old.(*api.Pod)
 	newPod.Spec = oldPod.Spec
 	newPod.DeletionTimestamp = nil
+
+	// don't allow the pods/status endpoint to touch owner references since old kubelets corrupt them in a way
+	// that breaks garbage collection
+	newPod.OwnerReferences = oldPod.OwnerReferences
```

Comment:
we need to keep this as-is.  guarding on termination is insufficient.  the older kubelets hit status update failures independent of the pod deletion problem when paired with this plug-in.